pkg/network/setup/network.go

```diff
@@ -80,6 +80,17 @@ func (v VMNetworkConfigurator) getPhase2NICs(domain *api.Domain, networks []v1.N
 }
 
 func (n *VMNetworkConfigurator) SetupPodNetworkPhase2(domain *api.Domain, networks []v1.Network) error {
+	ifaceStatusesByName := vmispec.IndexInterfaceStatusByName(n.vmi.Status.Interfaces,
+		func(ifaceStatus v1.VirtualMachineInstanceNetworkInterface) bool {
+			return vmispec.HasIfaceStatusOriginatedFromSpec(ifaceStatus) && ifaceStatus.PodInterfaceName != ""
+		},
+	)
+
+	networks = vmispec.FilterNetworksSpec(networks, func(network v1.Network) bool {
+		_, hasPodIfaceName := ifaceStatusesByName[network.Name]
+		return hasPodIfaceName
+	})
+
```

Comment:
Since there is already filtering logic on networks phase2, please consider embedding this logic in the existing logic (getPhase2NICs / newPhase2PodNIC).

Reply:
Please see the commit whose title is `virt-launcher, net setup: Skip networks with unknown PodInterfaceName`.

Reply:
> Please see the commit whose title is `virt-launcher, net setup: Skip networks with unknown PodInterfaceName`.

@orelmisan I am not sure what do I miss..
The filtering is done at a higher level, I am suggesting doing that in lower layers (getPhase2NICs / newPhase2PodNIC) because it already does some similar filtering.
Doing so can help centralized the filtering deing done in phase2, having it less diverged.

Not a must for this PR.